rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_publisher_info.hpp

```diff
@@ -146,6 +138,12 @@ class PubListener : public EventListenerInterface, public eprosima::fastdds::dds
   bool incompatible_qos_changes_
   RCPPUTILS_TSA_GUARDED_BY(on_new_event_m_);
 
+  eprosima::fastdds::dds::PublicationMatchedStatus matched_status_
+  RCPPUTILS_TSA_GUARDED_BY(on_new_event_m_);
+
+  eprosima::fastdds::dds::PublicationMatchedStatus unmatched_status_
```

Comment:
It's weird for me to define a `unmatched_status` with a structure named `PublicationMatchedStatus`.
Can we use the structures defined in the `rmw` directly?

Currently, there are two separate calculations in both `on_*_matched` and `take_event` for `unmatched_status`.
Can we just update the final data in `on_*_matched`?

Reply:
> Can we use the structures defined in the rmw directly?

Yes. Need not use eprosima::fastdds::dds::PublicationMatchedStatus.

> Currently, there are two separate calculations in both on_*_matched and take_event for unmatched_status.
> Can we just update the final data in on_*_matched?

Yes. I will optimize code.